stonesoup/updater/particle.py

```diff
@@ -83,13 +97,20 @@ def update(self, hypothesis, **kwargs):
         predicted_state.log_weight = new_weight
 
         # Resample
+        resample_flag = True
         if self.resampler is not None:
-            predicted_state = self.resampler.resample(predicted_state)
+            resampled_state = self.resampler.resample(predicted_state)
+            if resampled_state == predicted_state:
+                resample_flag = False
```

Comment:
There is a bug here if regulariser isn't defined. 

I think you should just set predicated state per suggestion, and then also remove `if` statement below.

```suggestion
                resample_flag = False
            predicted_state = resampled_state
```

Reply:
That is a problem. I will get that sorted in a new commit.